File: pokeR/ClientApp/src/app/models/entities/room.ts
import { Deck } from './deck';
import { User } from './user';

export class Room {
    public id: string;
    public name: string;
    public tagLine: string;
    public timeCreated: Date;
    public deckId: number;
    public deck: Deck;
    public users: Array<User>;
}

File: PokeR.Core/Entities/Room.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace PokeR.Core.Entities
{
    public class Room
    {
        [Key]
        public string Id { get; set; }
        [Required]
        public string Name { get; set; }
        public string TagLine { get; set; }
        [Required]
        public DateTime TimeCreated { get; set; }
        public int DeckId { get; set; }

        public virtual ICollection<User> Users { get; set; }
        public virtual Deck Deck { get; set; }
    }
}
